### PR TITLE
Add `OptimizerConfig` and `SchedulerConfig`

### PR DESCRIPTION
**Pull request by jon-tow**
This PR adds the following configuration updates:

- Adds an `OptimizerConfig` dedicated to customizing optimizer specifications.
    - Previously users were hard constrained to `torch.optim.AdamW`.
    - This should make it easier to add other optimizers like 8-bit optimizers
      introduced in PR #133

- Adds a `SchedulerConfig` for customizing the learning rate scheduler.
    - Previously users were hard constrained to `torch.optim.lr_scheduler.CosineAnnealingLR`; this opens possibilities for other schedulers like `torch.optim.lr_scheduler.OneCycleLR` if desired from the community 

Notes:

- This PR formats the config YAML files throughout the repository to be more
  consistent.

`wandb` reports: [PPO/ILQL Sentiments](https://wandb.ai/jon-tow/trlx/reports/trlx-Add-OptimizerConfig-and-SchedulerConfig-135--VmlldzozMTQ1NTQw?accessToken=b3quf5f60zz3quxch85w2p5avobeop5hb73sgydoadfsu89get5y0lykzeqjmv4p)

@ reviewer: I found these updates useful to me; if this feature is not desired or designed appropriately feel free to suggest edits or close.

### CODE REVIEW

File: trlx/utils/__init__.py [hunk 1]
```diff
@@ -86,6 +87,50 @@ def get_distributed_config(accelerator: Accelerator):
     return dist_config
 
 
+class OptimizerNames(Enum):
```
Comment (LouisCastricato) [inline trlx/utils/__init__.py:90]: Why are we not using registers...?

**Review by LouisCastricato — APPROVED** [review]

File: trlx/utils/__init__.py (hunk 1 above)
Comment (jon-tow) [reply trlx/utils/__init__.py:90]: Adding a registry for `str` types seemed like a pre-mature abstraction given that the source is local to a single file. What exactly are you suggesting?

Comment (LouisCastricato) [reply trlx/utils/__init__.py:90]: Fair, Dw then. Ready for merging imho.
